index.js

```diff
@@ -0,0 +1,124 @@
+class ConcurrencyQueue {
+
+  constructor(concurrency = 0) {
+    this.concurrency = concurrency;
+    this.queue = [];
+    this.count = 0;
+    this.stopped = true;
+  }
+
+  start() {
+    if (this.stopped) {
+      this.stopped = false;
+      this.run();
+    }
+  }
+
+  async push(task) {
+    this.queue.push(task);
+    this.start();
```

Comment:
there is no need to start here. What if you'd like to gather some tasks and only then trigger their execution?